### PR TITLE
[Serverless] Add runtime duration enhanced metric

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Add `aws.lambda.enhanced.runtime_duration` metric by : 
- subscribing to LogsAPI with the new schemaVersion
- saving invoke time 
- subtracting LogTypePlatformRuntimeDone timestamp to the previous saved time

### Motivation

Better customer experience

### Describe how to test your changes

Unit test + deployed a function with the latest extension build

Screen Shot 2021-09-21 at 3 34 26 PM

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/serverless/serverless.go [hunk 1]
```diff
@@ -172,9 +172,6 @@ func WaitForNextInvocation(stopCh chan struct{}, daemon *daemon.Daemon, id regis
 			metricTags := tags.AddColdStartTag(daemon.ExtraTags.Tags, daemon.ExecutionContext.Coldstart)
 			metricsChan := daemon.MetricAgent.GetMetricChannel()
 			metrics.SendTimeoutEnhancedMetric(metricTags, metricsChan)
-			if err != nil {
```
Comment (maxday) [inline pkg/serverless/serverless.go:175]: this is unrelated to this PR but this useless log should have been removed in v10 refactoring

File: pkg/serverless/logs/testdata/platform_incorrect_runtime_done_log_valid.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (nhinsch) [inline pkg/serverless/logs/testdata/platform_incorrect_runtime_done_log_valid.json:1]: I think this file name might be wrong -- it says "valid" but it also says "incorrect"

File: pkg/serverless/metrics/enhanced_metrics.go [hunk 3]
```diff
@@ -31,6 +32,25 @@ func getOutOfMemorySubstrings() []string {
...
 }
 
+// GenerateRuntimeDurationMetric generates the runtime duration metric
+func GenerateRuntimeDurationMetric(start time.Time, end time.Time, status string, tags []string, metricsChan chan []metrics.MetricSample) {
+	// first check if both date are set
+	if start.IsZero() || end.IsZero() {
+		log.Debug("Impossible to compute aws.lambda.enhanced.runtime_duration due to an invalid interval")
+	} else {
+		duration := end.Sub(start).Milliseconds()
+		// TODO do we want to do something with status (success/error/timeout) here ?
```
Comment (nhinsch) [inline pkg/serverless/metrics/enhanced_metrics.go:42]: Eventually I think we will want to generate the error metric here, but for now I would remove the TODO comment just to keep the codebase cleaner

**Review by nhinsch — APPROVED** [review]
Looks great! Just a couple minor nits

File: pkg/serverless/logs/logs.go [hunk 4]
```diff
@@ -49,61 +49,68 @@ type CollectionRouteInfo struct {
...
+	logTypePlatformEnd = "platform.end"
+	// logTypePlatformReport is used for the log messages containing a report of the last invocation.
+	logTypePlatformReport = "platform.report"
+	// logTypePlatformLogsDropped is used when AWS has dropped logs because we were unable to consume them fast enough.
+	logTypePlatformLogsDropped = "platform.logsDropped"
+	// logTypePlatformLogsSubscription is used for the log messages about Logs API registration
+	logTypePlatformLogsSubscription = "platform.logsSubscription"
+	// logTypePlatformExtension is used for the log messages about Extension API registration
+	logTypePlatformExtension = "platform.extension"
+	//lLogTypePlatformRuntimeDone is received when the runtime (customer's code) has returned (success or error)
```
Comment (nhinsch) [inline pkg/serverless/logs/logs.go:99]: Nit: This looks like a typo

**Pull request merged**